server/controllers/BasicServices.js

```diff
@@ -1,185 +1,279 @@
     })
-    .catch(function (response) {
-      utils.writeJson(res, response);
-      executionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, 500, req.body, responseBody);
+    .catch(async function (response) {
+      let responseCode = responseCodeEnum.code.INTERNAL_SERVER_ERROR;
+      let responseHeader = await restResponseHeader.createResponseHeader(xCorrelator, startTime, req.url);
+      responseBuilder.buildResponse(res, responseCode, response, responseHeader);
+      executionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, responseCode, req.body, response);
     });
 };
 
 module.exports.endSubscription = function endSubscription (req, res, next, body, user, originator, xCorrelator, traceIndicator, customerJourney) {
+  let startTime = process.hrtime();
+
   BasicServices.endSubscription(body, user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
-      executionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, 200, req.body, responseBody);
+    .then(async function (response) {
+        let responseCode = responseCodeEnum.code.NO_CONTENT;
+        let responseHeader = await restResponseHeader.createResponseHeader(xCorrelator, startTime, req.url);
+        responseBuilder.buildResponse(res, responseCode, response, responseHeader);
+        executionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, responseCode, req.body, response);
     })
-    .catch(function (response) {
-      utils.writeJson(res, response);
-      executionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, 500, req.body, responseBody);
+    .catch(async function (response) {
+        let responseCode = responseCodeEnum.code.INTERNAL_SERVER_ERROR;
+        let responseHeader = await restResponseHeader.createResponseHeader(xCorrelator, startTime, req.url);
+        responseBuilder.buildResponse(res, responseCode, response, responseHeader);
+        executionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, responseCode, req.body, response);
     });
 };
 
 module.exports.informAboutApplication = function informAboutApplication (req, res, next, user, originator, xCorrelator, traceIndicator, customerJourney) {
-  BasicServices.informAboutApplication(user, originator, xCorrelator, traceIndicator, customerJourney)
-    .then(function (response) {
-      utils.writeJson(res, response);
-      executionAndTraceService.recordServiceRequest(xCorrelator, traceIndicator, user, originator, req.url, 200, req.body, responseBody);
+  let startTime = process.hrtime();
+
+  BasicServices.informAboutApplication(body, user, originator, xCorrelator, traceIndicator, customerJourney)
```

Comment:
Yes, of course. This body parameter should not be here.
That was too much copy&pasting the code while I modified the source.

May I ask you how you did find this issue? I didn't find a way for the compiler to detect this  "ReferenceError: body is not defined". Executing the calls at runtime also does not work for me because of Authorization problems (HTTP-401) because of a missing operation-key.